homeassistant/components/alpha_vantage/sensor.py

```diff
@@ -110,48 +110,27 @@ class AlphaVantageSensor(SensorEntity):
     def __init__(self, timeseries, symbol):
         """Initialize the sensor."""
         self._symbol = symbol[CONF_SYMBOL]
-        self._name = symbol.get(CONF_NAME, self._symbol)
+        self._attr_name = symbol.get(CONF_NAME, self._symbol)
         self._timeseries = timeseries
-        self.values = None
-        self._unit_of_measurement = symbol.get(CONF_CURRENCY, self._symbol)
-        self._icon = ICONS.get(symbol.get(CONF_CURRENCY, "USD"))
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self.values["1. open"]
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes."""
-        if self.values is not None:
-            return {
-                ATTR_ATTRIBUTION: ATTRIBUTION,
-                ATTR_CLOSE: self.values["4. close"],
-                ATTR_HIGH: self.values["2. high"],
-                ATTR_LOW: self.values["3. low"],
-            }
-
-    @property
-    def icon(self):
-        """Return the icon to use in the frontend, if any."""
-        return self._icon
+        self._attr_unit_of_measurement = symbol.get(CONF_CURRENCY, self._symbol)
+        self._attr_icon = ICONS.get(symbol.get(CONF_CURRENCY, "USD"))
 
     def update(self):
         """Get the latest data and updates the states."""
         _LOGGER.debug("Requesting new data for symbol %s", self._symbol)
         all_values, _ = self._timeseries.get_intraday(self._symbol)
-        self.values = next(iter(all_values.values()))
+        values = next(iter(all_values.values()))
+        self._attr_state = values["1. open"]
+        self._attr_extra_state_attributes = (
+            {
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+                ATTR_CLOSE: values["4. close"],
+                ATTR_HIGH: values["2. high"],
+                ATTR_LOW: values["3. low"],
+            }
+            if values is not None
```

Comment:
How come a `None` check is needed here but not here: `self._attr_state = values["1. open"]` just above?
Either the `None` check should be moved earlier, or it can be dropped.